common/src/device/provisioning/transport/DeviceRegistrationResult.cs

```diff
@@ -14,128 +14,52 @@ namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Models
-            string registrationId = default(string),
-            DateTime? createdDateTimeUtc = default(DateTime?),
-            string assignedHub = default(string),
-            string deviceId = default(string),
-            string status = default(string),
-            string substatus = default(string),
-            string generationId = default(string),
-            DateTime? lastUpdatedDateTimeUtc = default(DateTime?),
-            int? errorCode = default(int?),
-            string errorMessage = default(string),
-            string etag = default(string),
-            JRaw payload = default(JRaw))
-        {
-            Tpm = tpm;
-            X509 = x509;
-            SymmetricKey = symmetricKey;
-            RegistrationId = registrationId;
-            CreatedDateTimeUtc = createdDateTimeUtc;
-            AssignedHub = assignedHub;
-            DeviceId = deviceId;
-            Status = status;
-            Substatus = substatus;
-            GenerationId = generationId;
-            LastUpdatedDateTimeUtc = lastUpdatedDateTimeUtc;
-            ErrorCode = errorCode;
-            ErrorMessage = errorMessage;
-            Etag = etag;
-            Payload = payload;
-            CustomInit();
-        }
-
-        /// <summary>
-        /// An initialization method that performs custom operations like setting defaults
-        /// </summary>
-        partial void CustomInit();
-
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "tpm")]
         public TpmRegistrationResult Tpm { get; set; }
 
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "x509")]
         public X509RegistrationResult X509 { get; set; }
 
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "symmetricKey")]
         public SymmetricKeyRegistrationResult SymmetricKey { get; set; }
 
-
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "registrationId")]
         public string RegistrationId { get; set; }
 
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "createdDateTimeUtc")]
         public DateTime? CreatedDateTimeUtc { get; set; }
 
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "assignedHub")]
         public string AssignedHub { get; set; }
 
-        /// <summary>
-        /// </summary>
         [JsonProperty(PropertyName = "deviceId")]
         public string DeviceId { get; set; }
 
-        /// <summary>
-        /// Gets or sets possible values include: 'unassigned', 'assigning',
-        /// 'assigned', 'failed', 'disabled'
-        /// </summary>
         [JsonProperty(PropertyName = "status")]
         public string Status { get; set; }
 
-        /// <summary>
-        /// Possible values include: 'initialAssignment', 'deviceDataMigrated', 'deviceDataReset'
```

Comment:
I realize this is an internal class but some of these xml comments might be useful, why remove them?

Reply:
I'll add the status and substatus comments back in and put in a note for the certificate. Every other property seems self-explanatory, right?